src/common/QskRgbValue.h

```diff
@@ -235,4 +235,150 @@ namespace QskRgb
+            {
+                case '0':
+                    rgb = ( rgb << 4 ) | 0x0;
+                    break;
+
+                case '1':
+                    rgb = ( rgb << 4 ) | 0x1;
+                    break;
+
+                case '2':
+                    rgb = ( rgb << 4 ) | 0x2;
+                    break;
+
+                case '3':
+                    rgb = ( rgb << 4 ) | 0x3;
+                    break;
+
+                case '4':
+                    rgb = ( rgb << 4 ) | 0x4;
+                    break;
+
+                case '5':
+                    rgb = ( rgb << 4 ) | 0x5;
+                    break;
+
+                case '6':
+                    rgb = ( rgb << 4 ) | 0x6;
+                    break;
+
+                case '7':
+                    rgb = ( rgb << 4 ) | 0x7;
+                    break;
+
+                case '8':
+                    rgb = ( rgb << 4 ) | 0x8;
+                    break;
+
+                case '9':
+                    rgb = ( rgb << 4 ) | 0x9;
+                    break;
+
+                case 'A':
+                case 'a':
+                    rgb = ( rgb << 4 ) | 0xA;
+                    break;
+
+                case 'B':
+                case 'b':
+                    rgb = ( rgb << 4 ) | 0xB;
+                    break;
+
+                case 'C':
+                case 'c':
+                    rgb = ( rgb << 4 ) | 0xC;
+                    break;
+
+                case 'D':
+                case 'd':
+                    rgb = ( rgb << 4 ) | 0xD;
+                    break;
+
+                case 'E':
+                case 'e':
+                    rgb = ( rgb << 4 ) | 0xE;
+                    break;
+
+                case 'F':
+                case 'f':
+                    rgb = ( rgb << 4 ) | 0xF;
+                    break;
+
+                default:
+                    rgb = ( rgb << 4 ) | 0x0;
+                    break;
+            }
+        }
+
+        // add default 0xFF for alpha channel
+        if ( len == rgb_len )
+        {
+            rgb |= 0xFF000000;
+        }
+
+        // convert rrggbbaa to aarrggbb
+        if ( len == rgba_len )
+        {
+            rgb = ( 0x000000FF & rgb ) << 24 | ( rgb >> 8 );
+        }
+
+        return rgb;
+    }
+
+    namespace literals
+    {
+        QSK_EXPORT constexpr QRgb operator""_rgba(
+            const char* const str, const size_t len ) noexcept
+        {
+            return fromHexString( str, len );
+        }
+
+        QSK_EXPORT constexpr QColor operator""_color(
+            const char* const str, const size_t len ) noexcept
+        {
+          const auto argb = static_cast<int>(fromHexString( str, len ));
+          const auto r = ( argb >> 16) & 0xFF;
+          const auto g = ( argb >> 8) & 0xFF;
+          const auto b = ( argb >> 0) & 0xFF;
+          const auto a = ( argb >> 24) & 0xFF;
+          const auto color = QColor{r,g,b,a};
+          return color;
+        }
```

Comment:
```cpp
    {
        using namespace QskRgb::literals::qrgb;
        Q_ASSERT( "#112233AA"_rgba == 0xAA112233 );
        Q_ASSERT( "#112233AA"_argb == 0x112233AA );
    }

    {
        using namespace QskRgb::literals::color;
        Q_ASSERT( "#112233AA"_rgba == QColor::fromRgb( 0x11, 0x22, 0x33, 0xAA ) );
        Q_ASSERT( "#112233AA"_argb == QColor::fromRgb( 0x22, 0x33, 0xAA, 0x11 ) );
    }
```

Reply:
QColor is very present in the Qt API, so that it needs to be supported in the public API of QSkinny, However I'm thinking since quite some time of replacing it by QRgb in many places especially in QskGradientStop. Actually I'm even tempted to introduce our own color class - something like this:

```
class QskColor
{
    ...
    private:
        QRgb m_rgba;
};
```

This way it would be possible to have automatic conversions from QRgb/QColor/Qt::GlobalColor. It would also avoid the painful "QColor(QRgb rgb)" trap, where the alpha value gets lost - usually by accident.

Reply:
I mean now we have both ""_rgba/argb -> QRgb + QColor, and the user can/must use the suffix that defines the channel order in the string literal. One more measure to make to user think about it. Additionally we can now profit from the QColor constexpr C-TOR, so now we really have the flexibility and best of both world.